benchmark_test.go

```diff
@@ -218,3 +219,90 @@ func TestBenchmarkStatsPerPeer(t *testing.T) {
 	})
 	assert.Equal(t, want, statsServer.Aggregated(), "unexpected stats")
 }
+
+func TestBenchmarkOutput(t *testing.T) {
+	// Testing both plaintext and JSON output
+	tests := []struct {
+		name          string
+		opts          BenchmarkOptions
+		wantOutput    []string
+		notWantOutput []string
+	}{
+		{
+			name: "Test json output",
+			opts: BenchmarkOptions{
+				Format: "json",
+			},
+			wantOutput:    []string{"summary", "benchmarkParameters", "maxRPS"},
+			notWantOutput: []string{"Errors", "Benchmark parameters", "Max RPS"},
+		},
+		{
+			name: "Test plaintext output",
+			opts: BenchmarkOptions{
+				Format: "plaintext",
+			},
+			wantOutput:    []string{"Benchmark parameters", "Max RPS"},
+			notWantOutput: []string{"Errors", "summary", "maxRPS"},
+		},
+		// unimplemented format should default to plaintext
+		{
+			name: "Test undefined output",
+			opts: BenchmarkOptions{
+				Format: "csv",
+			},
+			wantOutput:    []string{"Benchmark parameters", "Max RPS"},
+			notWantOutput: []string{"Errors", "benchmarkParameters", "maxRPS"},
+		},
+	}
+
+	var requests atomic.Int32
+	s := newServer(t)
+	defer s.shutdown()
+	s.register(fooMethod, methods.errorIf(func() bool {
+		requests.Inc()
+		return false
+	}))
+	m := benchmarkMethodForTest(t, fooMethod, transport.TChannel)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			requests.Store(0)
+			buf, _, out := getOutput(t)
+			opts := Options{
+				BOpts: BenchmarkOptions{
+					MaxRequests: 100,
+					MaxDuration: 100 * time.Second,
+					RPS:         120,
+					Connections: 50,
+					Concurrency: 2,
+					Format:      tt.opts.Format,
+				},
+				TOpts: s.transportOpts(),
+			}
+
+			runBenchmark(out, _testLogger, opts, _resolvedTChannelThrift, m)
+			bufStr := buf.String()
+
+			for _, want := range tt.wantOutput {
+				assert.Contains(t, bufStr, want)
+			}
+			for _, notWant := range tt.notWantOutput {
+				assert.NotContains(t, bufStr, notWant)
+			}
+
+			if opts.BOpts.Format == "json" {
+				// Creating struct from string of JSON output
+				var benchmarkOutput BenchmarkOutput
+				b := []byte(bufStr)
+				err := json.Unmarshal(b, &benchmarkOutput)
+				if err != nil {
+					fmt.Println("error:", err)
```

Comment:
we should validate there is no error instead of simply printing it.

Reply:
Adding return after to make sure no further code is executed if error is thrown.